### ISSUE TITLE
[Feature Proposal] Adding efficient backbone network, VoVNet

### ISSUE THREAD

**Issue by youngwanLEE** [facebookresearch/detectron2#646]
## 🚀 Feature

More efficient **backbone network** called ***[VoVNet](https://github.com/youngwanLEE/detectron2/tree/vovnet/projects/VoVNet)*** for detectron2.
It is implemented in [detectron2/project](https://github.com/facebookresearch/detectron2/tree/master/projects) as a extension form.

## Motivation

It is found that [VoVNet](https://arxiv.org/abs/1904.09730) shows better performance and faster speed than ResNet or DenseNet as a backbone network for object detection or segmentation.

In addition, we have tried to improve VoVNetV1 and developed VoVNetV2 that is adopted by [CenterMask](https://github.com/youngwanLEE/CenterMask).

The CenterMask project already demonstrates VoVNetV2  models outperform ResNet or ResNeXt backbone at faster speed.

So I think our VoVNetV2 is helpful for other researchers or developers in the computer vision community. 

Can I add this [VoVNet project](https://github.com/youngwanLEE/detectron2/tree/vovnet/projects/VoVNet) into the [detectron2/project](https://github.com/facebookresearch/detectron2/tree/master/projects) directory?

## Implementation details

We simply replace ResNe(X)t with VoVNet in Faster/Mask R-CNN, using the same hyper-parameters.

The VoVNet backbone was pre-trained on ImageNet dataset.

We already make the [VoVNet project](https://github.com/youngwanLEE/detectron2/tree/vovnet/projects/VoVNet) lints with `./dev/linter.sh`.

**Comment by ppwwyyxx** [comment]
The project sounds interesting and thanks for creating a detectron2 implementation for it.

However, the goal of having "projects" directory is to demonstrate that projects can be maintained separately, i.e. they don't need to be part of detectron2 to work with detectron2. In fact, it will be much easier for both you and us to maintain your project if you release it as a separate repo managed by you, that contains only the "VoVNet" directory (but not an entire detectron2 replicate). Some more research projects from Facebook will also be released in this way.

**Comment by breznak** [comment]
Thank you for making VoVnet, such lightweight model is really useful for practical applications! 
In my opinion it's a pity the backbone hasn't been mainlined but the usage you demonstrate in your repo is quite convenient still. 

As I understand it, adding your model is a matter of mostly publishing its configs in model ZOO?
